### DISCUSSION TITLE
Websockets on GCP

### DISCUSSION THREAD

**Discussion by deswif** [serverpod/serverpod#1025] [Q&A] [answered]
I deployed the server in gcp by terraform, but sockets do not work. the connection is terminated after the time specified by the timeout in the load balancer. I did not find anything on the Internet about this. How to deal with it?

**Comment by Isakdl** [comment, marked as answer]
Ah! This can be configured in the google cloud console. You still need to have a timeout set in the load balancer, but I suggest you put a high enough value that would make sense for your use-case. 

We do not have a setting for this in our Terraform scripts currently, ill create an issue on this and make sure we add this, as that is the most convenient way to achieve what you want to do.

In the meanwhile, you can configure this manually like this:

1. Go to https://console.cloud.google.com
2. Select your GCP project
3. In the main menu go to "Network services"
4. Click on "Load balancing"
5. You should see three tabs "Load balancers", "backends", "frontends", make sure "**Load balancers**" is selected.
6. Click on your loadbalancer in the list.
<img width="1203" alt="Screenshot 2023-06-22 at 14 10 45" src="https://github.com/serverpod/serverpod/assets/7395515/52c074b7-b218-4260-bf5f-357a27cdfe51">

7. Select "**Backend configuration**" then click the edit symbol of the "...-backend-api" service. 
<img width="1452" alt="Screenshot 2023-06-22 at 14 06 01" src="https://github.com/serverpod/serverpod/assets/7395515/6e10d9e1-94f1-4cab-9e62-0e91763ee9ec">

8. Edit the timeout to a suitable value for your application.
<img width="1167" alt="Screenshot 2023-06-22 at 14 07 24" src="https://github.com/serverpod/serverpod/assets/7395515/19ec2191-cda1-4d25-b1d5-0445217c6c27">

9. Click "Update".

10. Click "Update" (again after the popup).

**Reply by deswif** [reply]
I have set timeout to limited value and it still doesn't work(

**Reply by vlidholt** [reply]
Did you set it on the correct backend service (the api)? I just tried it for our multiplayer game demo and it worked for me.

**Reply by Tristannos** [reply]
@vlidholt Okay, but this is a temporary fix, what is the long run fix? Because it is really annoying if the app doen't update when it needs to.

**Reply by vlidholt** [reply]
This is not a temporary fix. It should be working. :)
